File: exemplo-script-alteracao-senha.py
# 🔐 SCRIPT DE ALTERAÇÃO DE SENHA DO ADMIN (Python)
# Este provavelmente foi o método usado

from supabase import create_client, Client
import os

# Configuração do Supabase
SUPABASE_URL = "https://[seu-projeto].supabase.co"
SUPABASE_KEY = "[sua-anon-key]"

supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)

def alterar_senha_admin():
    print("🔐 Iniciando alteração de senha do admin...")
    
    # 1. Fazer login com a senha antiga
    print("1️⃣ Fazendo login...")
    try:
        auth_response = supabase.auth.sign_in_with_password({
            "email": "admin@ciclik.com.br",
            "password": "Admin@123456"  # Senha antiga (fraca)
        })
        print("✅ Login bem-sucedido!")
    except Exception as e:
        print(f"❌ Erro no login: {e}")
        return
    
    # 2. Alterar para nova senha forte
    print("2️⃣ Alterando senha...")
    try:
        update_response = supabase.auth.update_user({
            "password": "Nova_Senha_Forte_Aqui_123!@#"
        })
        print("✅ Senha alterada com sucesso!")
        print("📧 Nova senha: Nova_Senha_Forte_Aqui_123!@#")
        print("⚠️ Guarde esta senha em local seguro!")
    except Exception as e:
        print(f"❌ Erro ao alterar senha: {e}")
        return
    
    # 3. Fazer logout
    supabase.auth.sign_out()
    print("🚪 Logout realizado")

if __name__ == "__main__":
    alterar_senha_admin()
    print("✅ Processo concluído!")

"""
CARACTERÍSTICAS DESTE MÉTODO:
- Login registrado em last_sign_in_at ✅
- Alteração imediata (milissegundos) ✅
- Sem necessidade de interface gráfica ✅
- Pode ser rodado de qualquer lugar ✅
- Explica os timestamps idênticos ✅

COMO RODAR:
1. Salvar como: alterar_senha_admin.py
2. pip install supabase
3. python alterar_senha_admin.py

ESTE SCRIPT EXPLICA:
- Por que houve login (last_sign_in_at atualizado)
- Por que foi em 3ms (código sequencial)
- Por que não há logs detalhados (via client, não admin)
- Por que não precisa de UI (script direto)
"""

File: exemplo-script-alteracao-senha.js
// 🔐 SCRIPT DE ALTERAÇÃO DE SENHA DO ADMIN
// Este provavelmente foi o método usado

import { createClient } from '@supabase/supabase-js'

// Configuração do Supabase
const supabaseUrl = 'https://[seu-projeto].supabase.co'
const supabaseAnonKey = '[sua-anon-key]'

const supabase = createClient(supabaseUrl, supabaseAnonKey)

async function alterarSenhaAdmin() {
  console.log('🔐 Iniciando alteração de senha do admin...')
  
  // 1. Fazer login com a senha antiga
  console.log('1️⃣ Fazendo login...')
  const { data: loginData, error: loginError } = await supabase.auth.signInWithPassword({
    email: 'admin@ciclik.com.br',
    password: 'Admin@123456' // Senha antiga (fraca)
  })
  
  if (loginError) {
    console.error('❌ Erro no login:', loginError.message)
    return
  }
  
  console.log('✅ Login bem-sucedido!')
  
  // 2. Alterar para nova senha forte
  console.log('2️⃣ Alterando senha...')
  const { data: updateData, error: updateError } = await supabase.auth.updateUser({
    password: 'Nova_Senha_Forte_Aqui_123!@#'
  })
  
  if (updateError) {
    console.error('❌ Erro ao alterar senha:', updateError.message)
    return
  }
  
  console.log('✅ Senha alterada com sucesso!')
  console.log('📧 Nova senha: Nova_Senha_Forte_Aqui_123!@#')
  console.log('⚠️ Guarde esta senha em local seguro!')
  
  // 3. Fazer logout
  await supabase.auth.signOut()
  console.log('🚪 Logout realizado')
}

// Executar
alterarSenhaAdmin()
  .then(() => console.log('✅ Processo concluído!'))
  .catch(err => console.error('❌ Erro:', err))

/* 
CARACTERÍSTICAS DESTE MÉTODO:
- Login registrado em last_sign_in_at ✅
- Alteração imediata (milissegundos) ✅
- Sem necessidade de interface gráfica ✅
- Pode ser rodado de qualquer lugar ✅
- Explica os timestamps idênticos ✅

COMO RODAR:
1. Salvar como: alterar-senha-admin.js
2. npm install @supabase/supabase-js
3. node alterar-senha-admin.js

OU VIA PYTHON:
1. pip install supabase
2. python alterar_senha_admin.py
*/
